src/test/java/org/pkcs11/jacknji11/CryptokiTest.java

```diff
@@ -225,36 +221,40 @@ public void testFindObjects() {
 
 
     public void testEncryptDecrypt() {
-        long session = CE.OpenSession(TESTSLOT);
+        long session = CE.OpenSession(TESTSLOT, CK_SESSION_INFO.CKF_RW_SESSION | CK_SESSION_INFO.CKF_SERIAL_SESSION, null, null);
         CE.LoginUser(session, USER_PIN);
 
-        long des3key = CE.GenerateKey(session, new CKM(CKM.DES3_KEY_GEN),
-                new CKA(CKA.VALUE_LEN, 24),
+        long aeskey = CE.GenerateKey(session, new CKM(CKM.AES_KEY_GEN),
+                new CKA(CKA.VALUE_LEN, 32),
                 new CKA(CKA.LABEL, "label"),
+                new CKA(CKA.ID, "label"),
```

Comment:
Does softhsm require these extra attributes you are adding here and in other functions?  I would prefer to keep to only those required.

Reply:
Different HSMs have different requirement, there are even some (AWS CloudHSM/Cavium) that requires CKA_ID to the set at generation time, albeit most HSMs do not require this. I added java comments.